hugegraph-core/src/main/java/com/baidu/hugegraph/auth/HugeProject.java

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright 2017 HugeGraph Authors
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.baidu.hugegraph.auth;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.tinkerpop.gremlin.structure.Graph;
+import org.apache.tinkerpop.gremlin.structure.T;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+
+import com.baidu.hugegraph.HugeGraphParams;
+import com.baidu.hugegraph.auth.SchemaDefine.Entity;
+import com.baidu.hugegraph.backend.id.Id;
+import com.baidu.hugegraph.backend.id.IdGenerator;
+import com.baidu.hugegraph.schema.VertexLabel;
+import com.baidu.hugegraph.type.define.Cardinality;
+import com.baidu.hugegraph.type.define.DataType;
+import com.baidu.hugegraph.util.E;
+import com.google.common.base.Strings;
+
+public class HugeProject extends Entity {
+
+    private String name;
+    private String desc;
+    private Id adminGroupId;
+    private Id opGroupId;
+    private Set<String> graphs;
+    private Id targetId;
+
+    public HugeProject(Id id) {
+        this(id, null, null, null, null, null, null);
+    }
+
+    public HugeProject(String name) {
+        this(name, null);
+    }
+
+    public HugeProject(String name, String desc) {
+        this(null, name, desc, null, null, null, null);
+    }
+
+    public HugeProject(Id id, String name, String desc, Id adminGroupId,
+                       Id opGroupId, Set<String> graphs, Id targetId) {
+        this.name = name;
+        this.desc = desc;
+        this.adminGroupId = adminGroupId;
+        this.opGroupId = opGroupId;
+        this.graphs = graphs;
+        this.id = id;
+        this.targetId = targetId;
+    }
+
+    @Override
+    public ResourceType type() {
+        return ResourceType.PROJECT;
+    }
+
+    @Override
+    public String label() {
+        return P.PROJECT;
+    }
+
+    public String opGroupName() {
+        return "op_" + this.name;
+    }
+
+    public String adminGroupName() {
+        return "admin_" + this.name;
+    }
```

Comment:
delete these methods

Reply:
The purpose of this method is to get "groupName", because it is related to project, so it is reasonable to be a member method of project, otherwise it is also to write a method similar to utils or wrap

Reply:
can move into createProject() method